packaging/macos/checkmono.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+
+//
+// NOTE: Mono.framework only ships intel dylibs, so cannot be loaded by the arm64 slice of the Launcher utility.
+// Splitting checkmono into its own intel-only utility allows it to be called through rosetta if the user
+// wants to force the game to run under mono-through-rosetta.
+//
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+
+#define SYSTEM_MONO_PATH "/Library/Frameworks/Mono.framework/Versions/Current/"
+#define SYSTEM_MONO_MIN_VERSION "6.4"
+
+typedef char *(* mono_get_runtime_build_info)(void);
+
+static int check_mono_version(const char *version, const char *req_version)
+{
+	char *req_end, *end;
+	long req_val, val;
+
+	while (*req_version)
+	{
+		req_val = strtol(req_version, &req_end, 10);
+		if (req_version == req_end || (*req_end && *req_end != '.'))
+		{
+			fprintf(stderr, "Bad version requirement string '%s'\n", req_end);
+			return 1;
+		}
+
+		req_version = req_end;
+		if (*req_version)
+			req_version++;
+
+		val = strtol (version, &end, 10);
+		if (version == end || val < req_val)
+			return 1;
+
+		if (val > req_val)
+			return 0;
+
+		if (*req_version == '.' && *end != '.')
```

Comment:
You already incremented `req_version` above, so the first conditional here is probably never true? If so we should move
```
		req_version = req_end;
		if (*req_version)
			req_version++;
```
below this check.

Reply:
Likewise here.